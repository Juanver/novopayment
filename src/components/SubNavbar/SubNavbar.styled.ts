import styled, { css } from "styled-components";
import { Spacing } from "../../assets/spacing/Spacing";

const ContainerSubNavbar = styled.div`
  display: flex;
  gap: ${Spacing.Spacing3};
  padding: ${Spacing.Spacing0} ${Spacing.Spacing4};
`;

const ContainerTextSubNavbar = styled.div`
  padding: ${Spacing.Spacing1};
`;

const TextSubNavbar = styled.p`
  margin: ${Spacing.Spacing0};
`;

export const StyledSubNavbar = {
  ContainerSubNavbar,
  ContainerTextSubNavbar,
  TextSubNavbar,
};
