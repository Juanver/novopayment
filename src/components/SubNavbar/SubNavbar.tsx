import React, { useEffect, useState } from 'react';
import { List } from '../../domain/list';
import { getList } from '../../proxy/list';
import { StyledSubNavbar } from './SubNavbar.styled';


export const SubNavbar: React.FC = () => {
  const [lists, setLists] = useState<List[]>([])

  const getListReponse = async (): Promise<void> => {
    const listData = await getList();
    setLists(listData)
  }

  useEffect(() => {
    getListReponse();
  }, [])

  return (
    <>
      {
        lists && lists.map(({ cardList, enterprises, paysTransfer, support }, index) => (
          <StyledSubNavbar.ContainerSubNavbar key={index}>
            <StyledSubNavbar.ContainerTextSubNavbar>
              Lista de tarjetas
            </StyledSubNavbar.ContainerTextSubNavbar>
            <StyledSubNavbar.ContainerTextSubNavbar>
              Transferencias y pagos
            </StyledSubNavbar.ContainerTextSubNavbar>
            <StyledSubNavbar.ContainerTextSubNavbar>
              AtenciÃ³n al cliente
            </StyledSubNavbar.ContainerTextSubNavbar>
            <StyledSubNavbar.ContainerTextSubNavbar>
              Empresas
            </StyledSubNavbar.ContainerTextSubNavbar>
          </StyledSubNavbar.ContainerSubNavbar>
        ))
      }
    </>
  )
}
