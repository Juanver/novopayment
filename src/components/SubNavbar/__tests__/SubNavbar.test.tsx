import React from 'react'
import { render } from '@testing-library/react';
import { SubNavbar } from '../SubNavbar';

describe('SubNavbar component', () => {
    it('Should render', () => {
        const { container } = render(<SubNavbar />);
        expect(container).toMatchSnapshot();
    })
})