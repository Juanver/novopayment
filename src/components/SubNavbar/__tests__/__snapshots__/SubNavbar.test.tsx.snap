// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubNavbar component Should render 1`] = `<div />`;
